schema/home/home_page_fairs_module.js

```diff
@@ -0,0 +1,35 @@
+import Fair from "schema/fair"
+import { GraphQLList, GraphQLObjectType } from "graphql"
+
+export const HomePageFairsModuleType = new GraphQLObjectType({
+  name: "HomePageFairsModule",
+  fields: {
+    results: {
+      type: new GraphQLList(Fair.type),
+      resolve: (root, options, request, { rootValue: { fairsLoader } }) => {
+        // Check for all fairs that are currently running
+        const isEligible = { has_full_feature: true, sort: "-start_at" }
+        const gravityOptions = { ...options, ...isEligible, status: "running" }
+        return fairsLoader(gravityOptions).then(runningFairs => {
+          // If there are less than 8, get the most recent closed fairs
+          if (runningFairs.length < 8) {
+            gravityOptions.status = "closed"
+            const newOptions = { ...gravityOptions, size: 8 - runningFairs.length }
```

Comment:
If you’re gonna merge objects here anyways, then you can also override the `status` here instead of the line above:

```js
const closedFairsOptions = { ...gravityOptions, status: "closed", size: 8 - runningFairs.length }
```